Docs 5120 landing page for api products pa (#120)
mulesoft/docs-api-manager: modules/ROOT/pages/api-products-modifying-products.adoc

= Modifying API Products
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

If your business requirement changes, you might want to modify a product by adding new API instances to it. To cover new scenarios, you might want to add a new API instance. You can also version an existing instance to add customizations. If a service is no longer required, you can remove the respective API instance that supports the service.

The following tasks are covered in this section:

> * <<api-products-creating-products, Adding API instances to Products>>

Reviewer: This link is not being rendered correctly. I think it's missing the correct location
```suggestion
* <<adding-api-instances-to-products, Adding API instances to Products>>
```

Author: Done! Thanks!

* <<removing-api-instances-from-products, Removing API instances from a Product>>
* <<creating-new-versions-of-products, Creating new versions of a Product>>

[[adding-api-instances-to-products]]
== Adding API instances to Products

> You might want to add API intances to Products that belong in the same business group of organzations.

Reviewer: ```suggestion
You might want to add API instances to Products that belong in the same business group of organizations.
```

Author: Done! Thanks!

=== To add a new API instance to a Product:
. From the left menu in API Manager, click Products.

> . From the Products list on the right details pane, expand the Product to view the version.

Reviewer: ```suggestion
. From the Products list on the right details pane, expand the Product to view the available versions.
```

Author: Done! Thanks!

. Click the version of the Product to which you want to add the API instance.
. From the top-right corner of the window, click Edit.
The Adding API Instances window appears.
+
image::add-api-instances.png[Adding API Instances]
+
. If the API instance and the Product belong in the same business group or organization, select the environment, API instance, and version that you want to add to the Product.
. If the API instance and Product belong in different business groups:
.. Click Add Organization.
.. Select the organization or business group in which your API instance is available.
.. From the environment, select the API instance, and version that you want to add.
. Click Save.

> If your Product is published in Exchange, your Exchange asset reflects the  new API instance. To publish your product in Exchange, see xref:api-products-publishing-to-exchange.adoc[Publishing API Products to Exchange].

Reviewer: I would explain the different possible paths that can happen here when the Product Version has already been published

1) If the API instance added is related to an asset that is already linked to the Exchange product asset, then it will reflect the new addition

2) If the API Instance added belongs to a new asset that is not yet liked to the Exchange product asset, then users will be required to create a new version of the product

Author: Done! Thanks!

[[removing-api-instances-from-products]]
== Removing API instances from Products
If your business requirement changes, you might want to modify a product by removing an existing API instance.

=== To remove an API instance from a Product:
. From the left menu in API Manager, click Products.
. From the Products list on the right details pane, expand the Product to view the version.
. From the top-right corner, click Edit.

> . Click the garbage-can icon next to the API instance that you want to remove. 

Reviewer: The icon was changed for a cross now (same for the business group deletion)

Author: Done! Thanks!

. If you want to remove all the API instances from a specific organization or business group, click the garbage-can icon next to the organization name.
[PA: I don't see this option in my version of the product.]

> . Click OK.

Reviewer: We should explain (same as when adding the instances), that if the API Instance removed is the last API linked to an Asset, then users will be required to create a new version

Author: Done! Thanks!

[[creating-new-versions-of-products]]

> == Creating new versions of Products

Reviewer: I see we have an apart to explain situations where the user might need to create a new version. Should we place those scenarios here?

Author: Not sure what this means. Please explain.

> If you modify a prouduct to suit a changing business requirement, you might need to version it appropriately.

Reviewer: ```suggestion
If you modify a product to suit a changing business requirement, you might need to version it appropriately.
```

Author: Done! Thanks!

=== To create additional versions of a Product:

. From the left menu In API Manager, click Products. 
+
image::api-products-create-new-version.png[Creating New Version of Product]
+
. From the right details pane, click Add version.
The New version for Product appears.
. In the Version field, specify the version number that you want to create: for example, v1.1 or v2.0.

> . Add API instances in the Product. 

Reviewer: Should we comment that it will be pre-populated with the existing instances from the previous version?

Author: Done! Thanks!

For more information about how to add API instances in a Product, see Adding API Instances.
. Publish your new version of the Product in Exchange.

== See Also
* xref:api-products-landing-page.adoc[API Products]

>  * xref:api-products-creating-products.adoc[Creating Products]

Reviewer: ```suggestion
* xref:api-products-creating-products.adoc[Creating Products]
```

Author: Done! Thanks!
